File: src/test/java/com/raychenon/leetcode/ReverseIntegerTest.java
package com.raychenon.leetcode;

import org.junit.BeforeClass;
import org.junit.Test;

import static junit.framework.TestCase.assertEquals;

/**
 * User: raychenon
 * Date: 2019-08-25
 * https://leetcode.com/problems/reverse-integer/solution/
 */
public class ReverseIntegerTest {
    private static ReverseInteger reverseInteger;

    @BeforeClass
    public static void setUp() {
        reverseInteger = new ReverseInteger();
    }

    @Test
    public void testReversePositiveInteger() {
        assertEquals(321, reverseInteger.reverse(123));
    }

    @Test
    public void testReverseNegativeInteger() {
        assertEquals(-321, reverseInteger.reverse(-123));
    }

    @Test
    public void testReverseIntegerWith0Ending() {
        assertEquals(21, reverseInteger.reverse(120));
    }

    @Test
    public void testReverseLargeInteger() {
        assertEquals(2147483641, reverseInteger.reverse(1463847412));
    }

    @Test
    public void testReverseLargeNegativeInteger() {
        assertEquals(-2147483641, reverseInteger.reverse(-1463847412));
    }

    @Test
    public void testReverseVeryLargeInteger() {
        // Integer.MAX_VALUE = 0x7fffffff = 2147483647
        assertEquals(1463847412, reverseInteger.reverse(2147483641));
    }

    @Test
    public void testReverseVeryLargeNegativeInteger() {
        // Integer.MIN_VALUE = 0x80000000 = -2147483648
        assertEquals(-1463847412, reverseInteger.reverse(-2147483641));
    }
}

File: src/test/kotlin/com/raychenon/kotlin/leetcode/ReverseIntegerTest.kt
package com.raychenon.kotlin.leetcode

import org.junit.Assert
import org.junit.Test

/**
 * User: raychenon
 * Date: 2019-08-26
 * https://leetcode.com/problems/reverse-integer
 */
class ReverseIntegerTest {

    @Test
    fun testReversePositiveInteger() {
        Assert.assertEquals(321, ReverseInteger.reverse(123))
    }

    @Test
    fun testReverseNegativeInteger() {
        Assert.assertEquals(-321, ReverseInteger.reverse(-123))
    }

    @Test
    fun testReverseInteger() {
        Assert.assertEquals(21, ReverseInteger.reverse(120))
    }

    @Test
    fun testReverseLargeInteger() {
        Assert.assertEquals(2147483641, ReverseInteger.reverse(1463847412))
    }

    @Test
    fun testReverseLargeNegativeInteger() {
        Assert.assertEquals(-2147483641, ReverseInteger.reverse(-1463847412))
    }

    @Test
    fun testReverseVeryLargeInteger() {
        // Integer.MAX_VALUE = 0x7fffffff = 2147483647
        Assert.assertEquals(1463847412, ReverseInteger.reverse(2147483641))
    }

    @Test
    fun testReverseVeryLargeNegativeInteger() {
        // Integer.MIN_VALUE = 0x80000000 = -2147483648
        Assert.assertEquals(-1463847412, ReverseInteger.reverse(-2147483641))
    }
}